packages/edit-site/src/components/page-patterns/dataviews-patterns.js

```diff
@@ -0,0 +1,339 @@
+	const [ view, setView ] = useState( DEFAULT_VIEW );
+	const isUncategorizedThemePatterns =
+		type === PATTERN_TYPES.theme && categoryId === 'uncategorized';
+	const { patterns, isResolving } = usePatterns(
+		type,
+		isUncategorizedThemePatterns ? '' : categoryId,
+		{
+			search: view.search,
+			// syncStatus:
+			// 	deferredSyncedFilter === 'all'
+			// 		? undefined
+			// 		: deferredSyncedFilter,
+		}
+	);
+	const fields = useMemo(
+		() => [
+			{
+				header: __( 'Preview' ),
+				id: 'preview',
+				render: ( { item } ) => (
+					<Preview item={ item } viewType={ view.type } />
+				),
+				minWidth: 120,
+				maxWidth: 120,
+				enableSorting: false,
+				enableHiding: false,
+			},
+			{
+				header: __( 'Title' ),
+				id: 'title',
+				getValue: ( { item } ) => item.title,
+				render: ( { item } ) => (
+					<Title
+						item={ item }
+						onClick={ () => {} }
+						categoryId={ categoryId }
+					/>
+				),
+				maxWidth: 400,
+				enableHiding: false,
+			},
+		],
+		[ view.type, categoryId ]
+	);
+
+	const { data, paginationInfo } = useMemo( () => {
+		if ( ! patterns ) {
+			return {
+				data: EMPTY_ARRAY,
+				paginationInfo: { totalItems: 0, totalPages: 0 },
+			};
+		}
+		let filteredData = [ ...patterns ];
+		// Handle filters.
+		if ( view.filters.length > 0 ) {
+			// view.filters.forEach( ( filter ) => {
+			// 	if (
+			// 		filter.field === 'author' &&
+			// 		filter.operator === OPERATOR_IN &&
+			// 		!! filter.value
+			// 	) {
+			// 		filteredData = filteredData.filter( ( item ) => {
+			// 			return item.author_text === filter.value;
+			// 		} );
+			// 	} else if (
+			// 		filter.field === 'author' &&
+			// 		filter.operator === OPERATOR_NOT_IN &&
+			// 		!! filter.value
+			// 	) {
+			// 		filteredData = filteredData.filter( ( item ) => {
+			// 			return item.author_text !== filter.value;
+			// 		} );
+			// 	}
+			// } );
+		}
+
+		// Handle sorting.
+		if ( view.sort ) {
+			const stringSortingFields = [ 'title' ];
+			const fieldId = view.sort.field;
+			if ( stringSortingFields.includes( fieldId ) ) {
+				const fieldToSort = fields.find( ( field ) => {
+					return field.id === fieldId;
+				} );
+				filteredData.sort( ( a, b ) => {
+					const valueA = fieldToSort.getValue( { item: a } ) ?? '';
+					const valueB = fieldToSort.getValue( { item: b } ) ?? '';
+					return view.sort.direction === 'asc'
+						? valueA.localeCompare( valueB )
+						: valueB.localeCompare( valueA );
+				} );
+			}
+		}
+
+		// Handle pagination.
+		const start = ( view.page - 1 ) * view.perPage;
+		const totalItems = filteredData?.length || 0;
+		filteredData = filteredData?.slice( start, start + view.perPage );
+		return {
+			data: filteredData,
+			paginationInfo: {
+				totalItems,
+				totalPages: Math.ceil( totalItems / view.perPage ),
+			},
+		};
+	}, [ patterns, view, fields ] );
+
+	const actions = useMemo( () => [ renameAction, exportJSONaction ], [] );
+	const onChangeView = useCallback(
```

Comment:
At some point we should consider whether we could bundle this logic as well as it repeats itself across data views usage. We may be able to do so by adding the `defaultConfigPerViewType ` as a prop.

Also, it seems we're adding more and more "config" props, maybe at some point we could have a "settings" prop with a lot of these instead of keeping them top level.

Reply:
Yeah. I'm thinking of landing an initial PR without all the actions too(to reduce the size of this PR) and have some follow ups to simplify things a bit.